vllm/worker/model_runner.py

```diff
@@ -846,12 +846,12 @@ def capture_model(self, kv_caches: List[torch.Tensor]) -> None:
         logger.info(f"Graph capturing finished in {elapsed_time:.0f} secs.")
 
     def __del__(self) -> None:
-        # Delete the CUDA graphs before deleting the CuPy NCCL communicator.
+        # Delete the CUDA graphs before deleting the pynccl communicator.
         # NOTE(woosuk): This is necessary because otherwise deadlocks can
         # happen.
         # FIXME(woosuk): This is a bit hacky. Find a more robust solution.
         self.graph_runners.clear()
-        self.cupy_nccl_backend = None
```

Comment:
I think this part can be deleted as we remove CuPy?

Reply:
I don't know why this code was needed before. So I don't want to change it 🤣  If you have more context, we can discuss if we can delete this.

Reply:
When using CuPy + CUDA graphs, deadlocks happen when the CuPy backend is deleted before the CUDA graphs using it are deleted. I actually don't know the reason for this, but this doesn't happen when using NCCL through PyTorch probably because the NCCL communicator managed by PyTorch is deleted at the very end.

Reply:
@youkaichao could you check whether we can delete this? You can simply run `python examples/llm_engine_example.py -tp 2` and see if the process hangs.

Reply:
Okay, will have a try.

Reply:
BTW, I do agree with you that we can remove this code some time later. If you'd like to do so, could you please add `TODO` in the code?

Reply:
I tested the code (removing the above two lines) for 100 times `for i in {1..100}; do python3 examples/llm_engine_example.py -tp=2 && echo "Test passed $i times" || break; done` , and I don't see any deadlocks. This gives us confidence in removing the code later. I left a comment there to remove it in v0.4.1 .